src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/AnimationTextures.java

```diff
@@ -0,0 +1,65 @@
+package io.github.amerebagatelle.fabricskyboxes.util.object;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.util.Identifier;
+
+import java.util.List;
+
+public class AnimationTextures {
+    public static final Codec<AnimationTextures> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+            Identifier.CODEC.listOf().fieldOf("north").forGetter(AnimationTextures::getNorth),
+            Identifier.CODEC.listOf().fieldOf("south").forGetter(AnimationTextures::getSouth),
+            Identifier.CODEC.listOf().fieldOf("east").forGetter(AnimationTextures::getEast),
+            Identifier.CODEC.listOf().fieldOf("west").forGetter(AnimationTextures::getWest),
+            Identifier.CODEC.listOf().fieldOf("top").forGetter(AnimationTextures::getTop),
+            Identifier.CODEC.listOf().fieldOf("bottom").forGetter(AnimationTextures::getBottom)
+    ).apply(instance, AnimationTextures::new));
+    private final List<Identifier> north;
+    private final List<Identifier> south;
+    private final List<Identifier> east;
+    private final List<Identifier> west;
+    private final List<Identifier> top;
+    private final List<Identifier> bottom;
+    private final int size;
+
+    public AnimationTextures(List<Identifier> north, List<Identifier> south, List<Identifier> east, List<Identifier> west, List<Identifier> top, List<Identifier> bottom) {
+        this.north = north;
+        this.south = south;
+        this.east = east;
+        this.west = west;
+        this.top = top;
+        this.bottom = bottom;
+
+        assert north.size() == south.size() && south.size() == east.size() && east.size() == west.size() && west.size() == top.size() && top.size() == bottom.size();
```

Comment:
this (usually) won't run in production

Reply:
you don't need to check that the sizes of the lists are equal if you use a list of `Texture`